lib/matplotlib/collections.py

```diff
@@ -28,53 +28,38 @@
 })
 class Collection(artist.Artist, cm.ScalarMappable):
     """
-    Base class for Collections.  Must be subclassed to be usable.
+    Base class for Collections. Must be subclassed to be usable.
 
-    All properties in a collection must be sequences or scalars;
-    if scalars, they will be converted to sequences.  The
-    property of the ith element of the collection is::
+    A Collection represents a sequence of `~.patches.Patch` objects that the
+    library believes would be more efficient to draw together than
+    individually. For example, when a single path is being drawn repeatedly at
```

Comment:
Also, I hesitate to remove the wording about "may be" more efficient, since for most renderers, `draw_path_collection` is not necessarily any more efficient than individual `draw_path` calls (it just typically is whenever Collections are used correctly).

But I do see why you wanted to change it, the wording is pretty awkward, so I'll try changing it.

Reply:
You may be mostly right with `Patch`. I was thinking of `LineCollection` which rather mimics a list of `Line2D` and thus went for the more general `Artist`. This is all a bit messy.

Is it reasonable to stick to `Patch` and just mention in `LineCollection` that it's different?

Performance: I didn't check, but I assume the advantage of collections is that you save a lot of python function calls, similar to how numpy is faster than a list of floats.

Reply:
I'll update the docs to refer to patch and make sure that LineCollection is clear about how it's basically the same thing but for Line2D.

I think you're right about the performance in theory, but in practice, backend base still breaks up `draw_path_collection` into individual `draw_path` calls anyway, so you don't gain performance unless the backend supports it, which was probably the justification for the original language.

Reply:
Not sure if the backend is the bottleneck or creating the Artists, but in a quick test LineCollection is faster:
~~~
import matplotlib.pyplot as plt
from matplotlib.lines import Line2D
from matplotlib.collections import LineCollection
import numpy as np

xs = np.linspace(0, 1, 1000)
~~~

~~~
%%timeit
fig, ax = plt.subplots()
for x in xs:
    line = Line2D([x, x], [0, 1])
    ax.add_line(line)
~~~
> 445 ms ± 9.8 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

~~~
%%timeit
fig, ax = plt.subplots()

lc = LineCollection([[(x, 0), (x, 1)] for x in xs])
ax.add_collection(lc)
~~~
> 39.7 ms ± 1.13 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)